homeassistant/components/light/lw12wifi.py

```
@@ -0,0 +1,173 @@
+        self._rgb_color = rgb_color
+        self._brightness = brightness
+        self._state = None
+        self._effect = effect
+        self._transition_speed = transition
+
+    @property
+    def name(self):
+        """Return the display name of the controlled light."""
+        return self._name
+
+    @property
+    def brightness(self):
+        """Return the brightness of the light."""
+        return self._brightness
+
+    @property
+    def rgb_color(self):
+        """Read back the color of the light."""
+        return self._rgb_color
+
+    @property
+    def hs_color(self):
+        """Read back the hue-saturation of the light."""
+        return color_util.color_RGB_to_hs(*self._rgb_color)
+
+    @property
+    def effect(self):
+        """Return current light effect."""
+        if self._effect is None:
+            return None
+        return self._effect.replace('_', ' ').title()
+
+    @property
+    def is_on(self):
+        """Return true if light is on."""
+        return self._state
+
+    @property
+    def supported_features(self):
+        """Return a list of supported features."""
+        return SUPPORT_LW12
+
+    @property
+    def effect_list(self):
+        """Return a list of available effects.
+
+        Use the Enum element name for display.
+        """
+        import lw12
+        return [effect.name.replace('_', ' ').title()
+                for effect in lw12.LW12_EFFECT]
+
+    @property
+    def assumed_state(self) -> bool:
+        """Return True if unable to access real state of the entity."""
+        return True
+
+    def turn_on(self, **kwargs):
+        """Instruct the light to turn on."""
+        import lw12
+        self._light.light_on()
+        if self._effect is not None:
+            if self._effect.replace(' ', '_').upper() in lw12.LW12_EFFECT:
+                kwargs['effect'] = self._effect
+            else:
+                # Unknown effect: changing to no selected effect.
```

If it's an unknown effect, shouldn't we just `raise HomeAssistantError`. I'm not a huge fan of failing silently.